moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -261,29 +286,38 @@ void MotionPlanningFrame::fillStateSelectionOptions()
   const robot_model::JointModelGroup* jmg = robot_model->getJointModelGroup(group);
   if (jmg)
   {
-    ui_->start_state_selection->addItem(QString("<random valid>"));
-    ui_->start_state_selection->addItem(QString("<random>"));
-    ui_->start_state_selection->addItem(QString("<current>"));
-    ui_->start_state_selection->addItem(QString("<same as goal>"));
+    ui_->start_state_combo_box->addItem(QString("<random valid>"));
+    ui_->start_state_combo_box->addItem(QString("<random>"));
+    ui_->start_state_combo_box->addItem(QString("<current>"));
+    ui_->start_state_combo_box->addItem(QString("<same as goal>"));
 
-    ui_->goal_state_selection->addItem(QString("<random valid>"));
-    ui_->goal_state_selection->addItem(QString("<random>"));
-    ui_->goal_state_selection->addItem(QString("<current>"));
-    ui_->goal_state_selection->addItem(QString("<same as start>"));
+    ui_->goal_state_combo_box->addItem(QString("<random valid>"));
+    ui_->goal_state_combo_box->addItem(QString("<random>"));
+    ui_->goal_state_combo_box->addItem(QString("<current>"));
+    ui_->goal_state_combo_box->addItem(QString("<same as start>"));
 
     const std::vector<std::string>& known_states = jmg->getDefaultStateNames();
     if (!known_states.empty())
     {
-      ui_->start_state_selection->insertSeparator(ui_->start_state_selection->count());
-      ui_->goal_state_selection->insertSeparator(ui_->goal_state_selection->count());
+      ui_->start_state_combo_box->insertSeparator(ui_->start_state_combo_box->count());
+      ui_->goal_state_combo_box->insertSeparator(ui_->goal_state_combo_box->count());
       for (std::size_t i = 0; i < known_states.size(); ++i)
       {
-        ui_->start_state_selection->addItem(QString::fromStdString(known_states[i]));
-        ui_->goal_state_selection->addItem(QString::fromStdString(known_states[i]));
+        ui_->start_state_combo_box->addItem(QString::fromStdString(known_states[i]));
+        ui_->goal_state_combo_box->addItem(QString::fromStdString(known_states[i]));
       }
     }
-    ui_->start_state_selection->setCurrentIndex(2);  // default to 'current'
-    ui_->goal_state_selection->setCurrentIndex(0);   // default to 'random valid'
+
+    ui_->start_state_combo_box->blockSignals(false);
+    ui_->goal_state_combo_box->blockSignals(false);
+    ui_->start_state_combo_box->setCurrentIndex(2);                                // default to 'current'
+    ui_->goal_state_combo_box->setCurrentIndex(2);                                 // default to 'current'
+    ui_->planning_group_combo_box->setCurrentText(QString::fromStdString(group));  // currently chosen group
```

Comment:
Why do you need to touch the `planning_group_combo_box`?
Semantically this doesn't fit into `fillStateSelectionOptions()`.

Reply:
This is needed to update the combo box, whenever the planning group is changed via the display property.

I deleted this line and replaced the functionality by calling `fillPlanningGroupOptions()` in `changePlanningGroupHelper()` instead.
Let me know if you would prefer a seperate function that only chooses the correct planning group instead of retrieving the planning groups and refilling the combo box at each change.

Reply:
No. The `planning_group_combo_box` only needs to be updated once when the `RobotModel` was loaded. I implemented and pushed the corresponding fix to your PR branch. @2scholz, please have a final look and rebase onto current melodic-devel.

Reply:
It makes sense to only fill the options of the `planning_group_combo_box` once when the `RobotModel` is loaded.
But this breaks the synchronization between the property in the `MotionPlanningDisplay` and the drop-down menu in the panel. Whenever the planning group is changed via the property, the drop-down menu does not show the currently selected group anymore.

To fix this I added the method `setPlanningGroupText` to update the index of the `planning_group_combo_box`. It is used in the `changePlanningGroupHelper`.